docs/docs/05-reusable-components.md

```diff
@@ -248,7 +248,7 @@ ReactDOM.render(<HelloMessage name="Sebastian" />, mountNode);
 ```
 
 This simplified component API is intended for components that are pure functions of their props. These components must not retain internal state, do not have backing instances, and do not have the component lifecycle methods. They are pure functional transforms of their input, with zero boilerplate.
-However, you may still specify `.propTypes` and `.defaultProps` by setting them as properties on the function, just as you would set them on an ES6 class.
+However, you may still specify `.propTypes`, `.defaultProps`, and `.displayName` by setting them as properties on the function, just as you would set them on an ES6 class. If `.displayName` or `.name` (in that order) is a non-empty string it will be used as the name of the component.
```

Comment:
How about:
 > If `.displayName` or `.name` is a non-empty string it may be used as the name of the component.

Removes " (in that order)" and replaces "will" with "may".  The reason being that we are also going to start pulling name info from the jsx source transform, so we don't want to be making guarantees about how component names are derived.